lib/dotcom/alerts/disruptions/subway.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Dotcom.Alerts.Disruptions.Subway do
+  @moduledoc """
+  Disruptions are alerts that have `service_impacting_effects` grouped by `service_range`.
+  """
+
+  import Dotcom.Alerts, only: [service_impacting_alert?: 1]
+  import Dotcom.Routes, only: [subway_route_ids: 0]
+  import Dotcom.Utils.ServiceDateTime, only: [service_range: 1]
+
+  alias Alerts.Alert
+  alias Dotcom.Utils
+
+  @alerts_repo Application.compile_env!(:dotcom, :repo_modules)[:alerts]
+
+  @doc """
+  Disruptions that occur any time after today's service range.
+  """
+  @spec future_disruptions() :: %{Utils.ServiceDateTime.named_service_range() => [Alert.t()]}
+  def future_disruptions() do
+    disruption_groups() |> Map.take([:later_this_week, :next_week, :after_next_week])
+  end
+
+  @doc """
+  Disruptions that occur during today's service range.
+  """
+  @spec todays_disruptions() :: %{today: [Alert.t()]}
+  def todays_disruptions() do
+    disruption_groups() |> Map.take([:today])
+  end
+
+  # Groups all disruption alerts by service range.
+  #
+  # 1. Gets all alerts for subway routes.
+  # 2. Filters out non-service-impacting alerts
+  # 3. Groups them according to service range.
+  defp disruption_groups() do
+    subway_route_ids()
```

Comment:
I don't think this will work correctly (unless I missed a change in `by_route_ids/2`. Green line alerts aren't associated with the route ID `"Green"` - they're associated with the branches - `"Green-B"`/etc. This will also miss Mattapan alerts, I think.

Reply:
I was pulling Green Line alerts with this code. I can check to see if we want Mattapan alerts to show up in this component. Right now it only shows the main lines. Easy to add if we need/want them.